ovro_alert/lwa_alert_client.py

```diff
@@ -35,16 +40,37 @@ def trigger(self):
             		pipeline.triggered_dump.trigger(ntime_per_file=self.ntime_per_file, nfile=self.nfile, dump_path=path)
 
 
-    def powerbeam(self):
+    def powerbeam(self, dd2):
         """ Observe with power beam
         """
-        pass
+	args = dd2  # Use dd2 directly
+    	position = args["position"].split(",")  # Parse the position to get ra and dec
+
+    	RA = float(position[0])  
+    	Dec = float(position[1]) 
+
+	dm = dd2["dm"]
+	toa = dd2["toa"]
+
+	max_delay = 4.149*1e3 * dm * 12**(-2) # maximum delay for a specific value of DM at the lowest LWA frequency (12 MHz) in seconds
+
+
+        self.con.configure_xengine(recorders=['dr1'], calibratebeams=True)
+    	thread = threading.Thread(target=self.con.control_bf, kwargs={'num': 1, 'targetname': (RA, Dec), 'track': True})
+    	thread.start()
+	
+
+	#Observe for the duration equal to maximum delay (duration is in ms)	
+    	self.con.start_dr(recorders=['dr1'], duration=max_delay*1e3, time_avg=128)
 
+	# Sleep for the duration + 10 sec, then stop the recording and the xengine
+    	time.sleep(max_delay + 10)  
```

Comment:
Instead of a fixed time relative to the start, it may be better to tell the data recorder to record until `max_delay+10` relative to the CHIME event time. That is, if we start late, it should record less data (or none at all, if starting very late).